src/main/java/de/fraunhofer/aisec/cpg/passes/CallResolver.java

```diff
@@ -339,51 +340,175 @@ private boolean compatibleSignatures(List<Type> callSignature, List<Type> functi
     return new ArrayList<>();
   }
 
+  /**
+   * @param call CallExpression
+   * @param functionDeclaration FunctionDeclaration the CallExpression was resolved to
+   * @return list containing the signature containing all argument types including the default
+   *     arguments
+   */
+  private List<Type> getCallSignatureWithDefaults(
+      CallExpression call, FunctionDeclaration functionDeclaration) {
+    List<Type> callSignature = new ArrayList<>(call.getSignature());
+    if (call.getSignature().size() < functionDeclaration.getParameters().size()) {
+      callSignature.addAll(
+          functionDeclaration
+              .getDefaultParameterSignature()
+              .subList(
+                  call.getArguments().size(),
+                  functionDeclaration.getDefaultParameterSignature().size()));
+    }
+
+    return callSignature;
+  }
+
+  /**
+   * @param constructExpression ConstructExpression
+   * @param constructorDeclaration ConstructorDeclaration the ConstructExpression was resolved to
+   * @return list containing the signature containing all argument types including the default
+   *     arguments
+   */
+  private List<Type> getCallSignatureWithDefaults(
+      ConstructExpression constructExpression, ConstructorDeclaration constructorDeclaration) {
+    List<Type> callSignature = new ArrayList<>(constructExpression.getSignature());
+    if (constructExpression.getSignature().size() < constructorDeclaration.getParameters().size()) {
+      callSignature.addAll(
+          constructorDeclaration
+              .getDefaultParameterSignature()
+              .subList(
+                  constructExpression.getArguments().size(),
+                  constructorDeclaration.getDefaultParameterSignature().size()));
+    }
+
+    return callSignature;
+  }
+
+  /**
+   * Adds the implicit default arguments to the CallExpression if they were not provided
+   *
+   * @param functionDeclaration the CallExpression has been resolved to containing the default
+   *     arguments
+   * @param call CallExpression which does not contain all necessary arguments and uses the default
+   *     arguments
+   */
+  private void addDefaultArgsToCall(FunctionDeclaration functionDeclaration, CallExpression call) {
+    if (functionDeclaration.hasSignature(getCallSignatureWithDefaults(call, functionDeclaration))) {
+      for (Expression expression :
+          functionDeclaration
+              .getDefaultParameters()
+              .subList(
+                  call.getArguments().size(), functionDeclaration.getDefaultParameters().size())) {
```

Comment:
Why are we adding defaults as arguments to the call-expression itself? I would strictly oppose this if it is not necessary as the arguments are AST-Nodes. if we add the defaults to the arguments we will not be able to search the CallExpression by its syntax appearing in the original code and create potential loobs in subgraphs that connect arguments and Parameters/Initializers of Parameters.

Is there a good reason that cannot be solved in another way?

Reply:
We had this discussion (somewhere - not sure where exactly now). They way it works for most language, is that default arguments are resolved before the call and then implicitly added to the call (expression). The original syntax can be retrieved by omitting those arguments that have a `DEFAULT` property set on the edge.

I am not sure, if I follow your example w.r.t. loops. Can you give a concrete example?

Reply:
Yes, but that only works if someone knows that his call will get some added arguments. That may lead to unexpected results if someone does not know that he/she has to ignore does.

With regards to the loop it is not something that I don't know if it may be an issue.